src/jsc/jsc_rpc_network_transport.hpp

```diff
@@ -0,0 +1,71 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "sync/generic_network_transport.hpp"
+#include "js_types.hpp"
+#include "js_network_transport.hpp"
+#include <string>
+
+namespace realm {
+namespace rpc {
+
+/*
+struct RPCFetchRef {
+    static JSValueRef value;
+};
+*/
+
+/**
+ * Provides an implementation of GenericNetworkTransport for use when the library is loaded in a runtime which doesn't provide the APIs required to perform network requests directly.
```

Comment:
I don't see the need to implement this. we should be able to use the JSNetworkTransport.

Reply:
Can you provide an explanation to this? What makes you think it would be better to use the `JSNetworkTransport`? That is basically just a wrapper around some JavaScript code which won't be loaded in the case of us running in chrome debugger mode. I see no advantage in trying to polyfill the `_networkTransport` property from C++.

Reply:
The way I see it is we have JavaScriptNetworkTransport which is a generic class to do network requests. It should be possible to use it instead of defining a new one for the debug context. The way it works is it just needs a `global.Realm._networkTransport.fetchWithCallbacks` in order to work.

Reply:
That was my initial thought as well. However it turned out to be way easier to implement an alternative specialisation of `GenericNetworkTransport` than providing a polyfill for `fetchWithCallbacks` implemented in C++. I'm sure you would have reached the same conclusion. From your comment, I don't see sufficient value in changing this approach and I'll leave it at this.